notes/force-move-spec.md

```diff
@@ -0,0 +1,208 @@
+- **Challenge** if `finalizesAt < currentTime`
+  - implies that all other fields are not null
+- **Finalized** if `finalizesAt >= currentTime`
+  - implies that all other fields are not null
+
+## Method behaviours
+
+### ForceMove
+
+Call:
+
+`forceMove(State[] states, Signatures[] signatures, challengerSig)`
+
+Notes:
+
+Determine the channel from the `challengeState` (the final state in the array).
+
+Requirements:
+
+- States form a chain of valid transitions
+- Channel is in the Open mode
+- Signatures are valid for the states
+- ChallengerSig is valid
+- ChallengeTurnNum is greater than turnNumRecord
+
+Effects:
+
+- Sets turnNumRecord to the challengeTurnNum
+- Sets finalizesAt to currentTime + challengeInterval
+- Sets stateHash
+- Sets challengerAddress
+
+### Respond
+
+Call:
+
+`respond(State challengeState, State nextState, Signature nextStateSig)`
+
+Notes:
+
+Determine the channel from the `nextState`.
+
+Requirements:
+
+- challengeState matches stateHash
+- challengeState to nextState is a valid transition
+- channel is in Challenge mode
+
+Effects:
+
+- Clears challenge (by clearing finalizesAt, stateHash and challengerAddress)
+- Increases the turnNumRecord by 1
+
+### RespondFromAlternative
+
+Call:
+
+`respondFromAlternative(State[] states, Signatures[] signatures)`
+
+Notes:
+
+Determine the channel from the `responseState` (the final state in the array).
+
+Requirements:
+
+- States form a chain of valid transitions
+- Channel is in the Challenge mode
+- Signatures are valid for the states
+- The responseState turnNum > turnNumRecord
+
+Effects:
+
+- Clears challenge (by clearing finalizesAt, stateHash and challengerAddress)
+- Sets turnNumRecord to the turnNum of the responseState
+
+### Refute
+
+Call
+
+`refute(State refutationState, Signature sig)`
+
+Notes:
+
+Determine the channel from the `refutationState` (the final state in the array).
+
+Requirements:
+
+- Channel is in the Challenge mode
+- The refutationState's turnNum is greater than the turnNumRecord
+- Sig is the challenger's signature on the refutationState
+
+Effects:
+
+- Clears challenge (by clearing finalizesAt, stateHash and challengerAddress)
+
+### Conclude
+
+Call
+
+`conclude(States states, Signatures sigs)`
+
+Notes:
+
+Determine the channel from the `concludeState` (the first state in the array).
+
+Requirements:
+
+- Channel is in the Open mode or the Challenge mode
+- First state is a conclude state
+- States form a chain of valid transitions
+- Signatures are valid for states
+
+Effects:
+
+- Sets finalizesAt to current time
+- Sets challengeHash to the hash of the concludeState
+- [Optionally] Clears turnNumRecord
+- [Optionally] Clears challengerAddress
+
+### SignaturesValid
+
+Internal call
+
+`_signaturesValid(States states, Signatures sigs)`
+
+Requirements:
+
+- There is a signature for each participant:
+  - either on the state for which they are a mover
+  - or on a state that appears after that state in the array
```

Comment:
I would like more context about why this might be useful. If we just want to extend functionality to cover all n participants signing a single state that they all agree on, this seems unnecessarily complicated.